### DISCUSSION TITLE
Some merged branches cannot be deleted

### DISCUSSION THREAD

**Discussion by carlos-granados** [Behat/Behat#1498] [Q&A]
We have three branches which have been merged but cannot be deleted:
- `dependabot/github_actions/majors-8a9da2b879`
- `chore-readme-urls-https`
- `bug-token-length-assertion`

When you try to delete them, Github complains that "You cannot delete this protected branch"

Anyone knows what's going on and how to fix it? 🤔

**Comment by acoulton** [comment]
My guess is @everzet has configured the branch protection rule for all branches in this repo, rather than just master / 4.x / etc.

I would prefer to work directly in behat/behat rather than a fork if I'm making changes myself (and dependabot will definitely always work in this repo) so ideally we'd target the branch protection on specific branches / patterns.

At some point in the future it would probably be useful if @carlos-granados and I had access to update branch protection / repo settings (including e.g. actions secrets) ourselves - but understand @everzet if you'd rather wait a while before giving us more permissions.

**Reply by stof** [reply]
Personally, I'm always sending PRs from my fork (when not using the GitHub UI to make a change, as it does not allow me to force it to use my fork), which avoids publishing my feature branches as versions on packagist.org

**Reply by acoulton** [reply]
That's a good point on packagist - I'll use a fork for code changes.

These were both quick fixes from the GitHub UI, and it'd be good to be able to still do that in future. As you say I think that can't be done from a fork.

**Reply by acoulton** [reply]
Urgh - and now I've accidentally pushed here instead of my fork and can't delete - sorry :disappointed: 

We probably want to update the ruleset to restrict creating new branches that match the rules - it looks like at the moment github is only protecting update and delete.

**Reply by everzet** [reply]
Disabled branch creation for now. Happy to relax in the future if you decide it is better XP, but for the time being, I'm siding with @stof

**Reply by acoulton** [reply]
Great, thanks @everzet  - and I've realised of course I can still use the Github web editor for quick changes, I just have to go to my fork first and do the update there. That's fine for me.

**Comment by everzet** [comment]
Deleted the branches.
